### PR TITLE
feat: Improve phone number capture

### PR DESCRIPTION
**Pull request by nishamishra12**
### 🔗 Linked issue

https://energysage.atlassian.net/browse/IDR-57

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

- Used https://vuejs-tips.github.io/vue-the-mask/ library to add masking to the phone number field
- The phone number field would be in use at multiple places, so I have conditionally added the mask with new type.
- Added the masked phone number field in molecules (FormInput) section as well as the form-field-validation section in examples
- Used this https://stackoverflow.com/questions/51836641/put-a-vue-directive-on-condition as a reference to conditionally use masking without null error
### 🥼 Testing

- Tested locally

#### 🧐 Feedback Requested / Focus Areas

-

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.
- [ ] I have documented testing approach

### CODE REVIEW

File: es-vue-base/src/lib-components/EsFormInput.vue [hunk 1]
```diff
@@ -130,6 +142,11 @@ export default {
             type: Boolean,
             default: false,
         },
+        phoneMaskValue: {
+            type: String,
+            default: '(###)-###-####',
```
Comment (nathanielwarner) [inline es-vue-base/src/lib-components/EsFormInput.vue:147]: This should be `(###) ###-####` rather than `(###)-###-####`

**Review by nathanielwarner — CHANGES_REQUESTED** [review]
This looks generally good and will improve the user experience. There's one thing we should correct in order to have the right format for the phone number.

File: es-vue-base/src/lib-components/EsFormInput.vue (hunk 1 above)
Comment (nishamishra12) [reply es-vue-base/src/lib-components/EsFormInput.vue:147]: Based on the screenshot that Xuewei  provided for design it is (###)-###-###. However, I can make this like the one you suggested as that is general US Format and it makes sense for the default to a standard format.

**Review by nathanielwarner — APPROVED** [review]
LGTM! 🚀

**Pull request merged**